### DISCUSSION TITLE
Unable to Search for Strings Starting with Square Brackets Using IdeaVim's Find Function

### DISCUSSION THREAD

**Discussion by xinxianwu** [JetBrains/ideavim#911] [Q&A]
I recently encountered an issue with the "find" functionality in IdeaVim. Neither "find next" nor "find previous" can search for strings that start with square brackets. For example, as shown in the image below:

https://github.com/JetBrains/ideavim/assets/50643550/4e500f09-566b-4602-9a2d-acc129b3062a

**Comment by OptimusCrime** [comment]
You need to escape it:

```
/\[Test
```

Should work.

**Comment by citizenmatt** [comment]
As @OptimusCrime mentioned, it works if you escape the `[` character. However, that's not normal Vim behaviour. This is a regression in the new regex engine in IdeaVim which treats the `[` as the start of a character range, and fails when it doesn't get the closing `]` (it's expecting `[Test]` to search for any instances of `T`, `e`, `s` or `t`). Vim only treats this as a character range if it has the closing `]`, and treats it as a simple search string otherwise. See [VIM-3497](https://youtrack.jetbrains.com/issue/VIM-3497).
